File: bindings/python/examples/get_transaction.py
# Copyright (c) 2025 IOTA Stiftung
# SPDX-License-Identifier: Apache-2.0

from lib.iota_sdk import *

import asyncio


async def main():
    client = GraphQlClient.new_localnet()

    transactions = await client.transactions()
    if not transactions.data:
        raise Exception("No transactions found")
    digest = transactions.data[0].transaction.digest()

    signed_transaction = await client.transaction(digest)
    print(f"Signed Transaction: `{signed_transaction}`\n")

    transaction_effects = await client.transaction_effects(digest)
    print(f"Transaction Effects: `{transaction_effects}`\n")

    transaction_data_effects = await client.transaction_data_effects(digest)
    print(f"Transaction Data Effects: `{transaction_data_effects}`\n")


if __name__ == "__main__":
    asyncio.run(main())

File: bindings/kotlin/examples/GetTransaction.kt
// Copyright (c) 2025 IOTA Stiftung
// SPDX-License-Identifier: Apache-2.0

import iota_sdk.GraphQlClient
import kotlinx.coroutines.runBlocking

fun main() = runBlocking {
    try {
        val client = GraphQlClient.newLocalnet()

        val transactions = client.transactions()
        if (transactions.data.isEmpty()) {
            throw Exception("No transactions found")
        }
        val digest = transactions.data[0].transaction.digest()

        val signedTransaction = client.transaction(digest)
        println("Signed Transaction: ${signedTransaction?.toString()}\n")

        val transactionEffects = client.transactionEffects(digest)
        println("Transaction Effects: ${transactionEffects?.toString()}\n")

        val transactionDataEffects = client.transactionDataEffects(digest)
        println("Transaction Data Effects: ${transactionDataEffects?.toString()}\n")
    } catch (e: Exception) {
        e.printStackTrace()
        kotlin.system.exitProcess(1)
    }
}
